test/fail_compilation/test18295.d

```diff
@@ -0,0 +1,24 @@
+/*
+TEST_OUTPUT:
+---
+fail_compilation/test18295.d(14): Error: functions cannot return `scope test18295.C`
+---
+*/
+
+// https://issues.dlang.org/show_bug.cgi?id=18295
+
+// See test/compilable/test18295.d for the `-dip1000` version`
+
+scope class C { int i; }    // Notice the use of `scope` here
+
+C increment(C c)            // Not compiling under -dip1000, so this should emit an error
```

Comment:
The comment seems to be wrong.

Reply:
Right or wrong, that's the way it currently works.  All this test is doing is adding coverage to the existing implementation.  The `compilable/test18295.d` file is the test for the new functionality in this PR.

Reply:
This test has been merged in a separate PR (https://github.com/dlang/dmd/pull/7791), so we can get off this tangent.